core/server/master/src/main/java/alluxio/master/file/meta/TtlBucket.java

```diff
@@ -34,16 +35,18 @@ public final class TtlBucket implements Comparable<TtlBucket> {
    */
   private static long sTtlIntervalMs =
       Configuration.getMs(PropertyKey.MASTER_TTL_CHECKER_INTERVAL_MS);
+  public static final int DEFAULT_RETRY_ATTEMPTS = 5;
   /**
    * Each bucket has a time to live interval, this value is the start of the interval, interval
    * value is the same as the configuration of {@link PropertyKey#MASTER_TTL_CHECKER_INTERVAL_MS}.
    */
   private final long mTtlIntervalStartTimeMs;
   /**
-   * A collection of inodes whose ttl value is in the range of this bucket's interval. The mapping
-   * is from inode id to inode.
+   * A collection containing those inodes whose ttl value is
+   * in the range of this bucket's interval. The mapping
+   * is from inode id to the number of left retry to process.
    */
-  private final ConcurrentHashMap<Long, Inode> mInodes;
+  private final ConcurrentHashMap<Long, Integer> mInodeToRetryMap;
```

Comment:
This data structure is low in memory utilization, but considering the thread safety i don't really have a better proposal...
https://github.com/Alluxio/alluxio/issues/17028

Reply:
yeah i was actually trying to use primitive structs such as GNU libs, but can't bcos of license issue. As this is not on a critical path, and not saving the inode should cut off a lot of mem utils, will hold off for now.